docs/input processing.md

```diff
@@ -0,0 +1,40 @@
+# Input processing
+The input process is about processing any input into our internal model representation `CommonInputModel`. This most likely have to be done using different methods based on which kind of input it is.
+
+As of now two inputs are supported:
+- JSON Schema Draft 7
+- AsyncAPI version 2.0.0
+
+## Internal model representation
+
+<img src="./assets/class_diagram.png"
+     alt="Class diagram" />
+
+As seen on the class diagram the `InputProcessor` is our main point of entry for processing input data. It uses the defined other input processors (`AsyncAPIInputProcessor`, `JsonSchemaInputProcessor`, ...) by calling `shouldProcess` function of each and if the function returns true processing it using the `process` function. If no processes returns true it defaults to `JsonSchemaInputProcessor`. 
```

Comment:
I would suggest following https://sembr.org here. In this case, just by adding line breaks after periods everything becomes more readable for humans (especially for reviewing/editing).